src/SOS/SOS.Extensions/HostServices.cs

```diff
@@ -255,6 +256,25 @@ protected override void Destroy()
             catch (InvalidCastException)
             {
             }
+            try
+            {
+                // IMemoryRegionService is currently only implemented on top of DbgEng.  Since this
+                // code uses COM marshalling (which only works on a windows .Net runtime), we also
+                // check to make sure this is a Windows OS to be safe.
+                if (HostType == HostType.DbgEng && RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
Couldn't you use the IDebugClient in DebuggerServices for most of this code from the HostType check up to the cast to IDebugControl5? It would save some duplicate code.